components/sidebar/sidebar_channel/sidebar_channel_menu/sidebar_channel_menu.tsx

```diff
@@ -31,133 +31,191 @@ type Props = PropsFromRedux & OwnProps;
-    const [openUp, setOpenUp] = useState(false);
 
     const {formatMessage} = useIntl();
 
-    let markAsReadMenuItem: JSX.Element | null = null;
+    let markAsReadUnreadMenuItem: JSX.Element | null = null;
     if (props.isUnread) {
-        function handleMarkAsRead() {
+        function handleMarkAsRead(event: MouseEvent<HTMLLIElement>) {
+            event.preventDefault();
+
             props.markChannelAsRead(props.channel.id);
             trackEvent('ui', 'ui_sidebar_channel_menu_markAsRead');
         }
 
-        markAsReadMenuItem = (
-            <Menu.ItemAction
+        markAsReadUnreadMenuItem = (
+            <Menu.Item
                 id={`markAsRead-${props.channel.id}`}
                 onClick={handleMarkAsRead}
-                icon={<MarkAsUnreadIcon size={16}/>}
-                text={formatMessage({id: 'sidebar_left.sidebar_channel_menu.markAsRead', defaultMessage: 'Mark as Read'})}
+                leadingElement={<MarkAsUnreadIcon size={18}/>}
+                labels={
+                    <FormattedMessage
+                        id='sidebar_left.sidebar_channel_menu.markAsRead'
+                        defaultMessage='Mark as Read'
+                    />
+                }
             />
+
         );
-    }
+    } else {
+        function handleMarkAsUnread(event: MouseEvent<HTMLLIElement>) {
+            event.preventDefault();
 
-    let markAsUnreadMenuItem: JSX.Element | null = null;
-    if (!props.isUnread) {
-        function handleMarkAsUnread() {
             props.markMostRecentPostInChannelAsUnread(props.channel.id);
             trackEvent('ui', 'ui_sidebar_channel_menu_markAsUnread');
         }
 
-        markAsUnreadMenuItem = (
-            <Menu.ItemAction
+        markAsReadUnreadMenuItem = (
+            <Menu.Item
                 id={`markAsUnread-${props.channel.id}`}
                 onClick={handleMarkAsUnread}
-                icon={<MarkAsUnreadIcon size={16}/>}
-                text={formatMessage({id: 'sidebar_left.sidebar_channel_menu.markAsUnread', defaultMessage: 'Mark as Unread'})}
+                leadingElement={<MarkAsUnreadIcon size={18}/>}
+                labels={
+                    <FormattedMessage
+                        id='sidebar_left.sidebar_channel_menu.markAsUnread'
+                        defaultMessage='Mark as Unread'
+                    />}
             />
         );
     }
 
-    let favoriteMenuItem: JSX.Element | null = null;
+    let favoriteUnfavoriteMenuItem: JSX.Element | null = null;
     if (props.isFavorite) {
-        function handleUnfavoriteChannel() {
+        function handleUnfavoriteChannel(event: MouseEvent<HTMLLIElement>) {
+            event.preventDefault();
+
             props.unfavoriteChannel(props.channel.id);
             trackEvent('ui', 'ui_sidebar_channel_menu_unfavorite');
         }
 
-        favoriteMenuItem = (
-            <Menu.ItemAction
+        favoriteUnfavoriteMenuItem = (
+            <Menu.Item
```

Comment:
this whole condition looks like it can be changed to only alter the props.

I would much rather prefer a style like this:
```typescript
// handleUnfavoriteChannel and handleFavoriteChannel could be ideally consolidated
const makeToggleFavoriteChannelHandler = (isFavorite) => isFavorite ? handleUnfavoriteChannel : handleFavoriteChannel;
const favoriteMenuItemToggle = (
    <Menu.Item
        id={`unfavorite-${props.channel.id}`}
        onClick={makeToggleFavoriteChannelHandler(props.isFavorite)}
        leadingElement={props.isFavorite ? <StarIcon size={18}/> : <StarOutlineIcon size={18}/>}
        labels={props.isFavorite ? (
            <FormattedMessage
                id='sidebar_left.sidebar_channel_menu.unfavoriteChannel'
                defaultMessage='Unfavorite'
            />
        ) : (
            <FormattedMessage
                id='sidebar_left.sidebar_channel_menu.favoriteChannel'
                defaultMessage='Favorite'
            />
        )}
    />
);
```

Reply:
Hey, Thank you for the feedback, but i would like to leave this as is. Hope you don't mind.

Reply:
In this case I actually do mind ... the code is way more readable and you safe a considerable amount of lines

Reply:
I agree that it's less lines, but I actually disagree about it being easier to read. I'd actually argue that taking more lines for it makes it easier to read since you can clearly see the code for each case in its own section rather than needing to parse which parts of each line occur in each case. That's why I tend to rarely use ternary statements and prefer if statements for anything that's even slightly complex.

In this case in particular,  none of the props that we're passing into the `Menu.Item` are actually shared between the favourite/unfavourite menu items even though they have a similar function for end users

Reply:
I have a question here, does it make any difference performance-wise?